dialogue-apache-hc4-client/src/main/java/com/palantir/dialogue/hc4/ApacheHttpClientChannels.java

```diff
@@ -92,7 +93,17 @@ public static Channel createSingleUri(String uri, CloseableClient client) {
         return BlockingChannelAdapter.of(new ApacheHttpClientBlockingChannel(client.client, url(uri)));
     }
 
+    /**
+     * Prefer {@link #createCloseableHttpClient(ClientConfiguration, String)}.
+     *
+     * @deprecated Use the overload with a channel name.
+     */
+    @Deprecated
     public static CloseableClient createCloseableHttpClient(ClientConfiguration conf) {
```

Comment:
😢my nice clean signatures.

What if as an alternative, the ClientConfiguration passed in by WC contained a TaggedMetricRegistry which happened to add the channel-name tag to everything (akin to the VersionedTaggedMetricRegistry)?

Reply:
Our list of all connection pools would be added to every channel name, right?

Reply:
Ok so now that we don't have the statics, what do you think of this WC-controlled channel-name tag idea? Then we wouldn't even need to change the signature?

Reply:
That’s probably for the best, though consumers outside of witchcraft will need to know to do that as well, which isn’t likely to happen. That case is likely one-channel-per-application and fine without the extra data.